app/src/features/rules/screens/rulesList/components/RulesList/components/RuleSelectionList/RuleSelectionList.tsx

```diff
@@ -0,0 +1,116 @@
+import React from "react";
+import { RULE_DETAILS } from "./constants";
+import { Divider } from "antd";
+import { NavLink, useNavigate } from "react-router-dom";
+import PATHS from "config/constants/sub/paths";
+import { trackRuleCreationWorkflowStartedEvent } from "modules/analytics/events/common/rules";
+import { RuleType } from "types";
+import { useFeatureLimiter } from "hooks/featureLimiter/useFeatureLimiter";
+import { FeatureLimitType } from "hooks/featureLimiter/types";
+import { PremiumFeature } from "features/pricing";
+import { PremiumIcon } from "components/common/PremiumIcon";
+import { TooltipPlacement } from "antd/lib/tooltip";
+import { redirectToCreateNewRule } from "utils/RedirectionUtils";
+import { PaidFeatureNudgeViewedSource } from "modules/analytics/events/common/pricing";
+import "./RuleSelectionList.scss";
+
+export interface RuleSelectionListProps {
+  /** Analytics event source */
+  source?: string;
+  groupId?: string;
+  premiumPopoverPlacement?: TooltipPlacement;
+  callback?: () => void;
+
+  /** Premium icon analytics event source */
+  premiumIconSource: PaidFeatureNudgeViewedSource;
+  onRuleItemClick?: (ruleType: RuleType) => void;
+}
+
+export const RuleSelectionList: React.FC<RuleSelectionListProps> = ({
+  source = "my_rules",
+  groupId,
+  premiumPopoverPlacement = "topLeft",
+  callback = () => {},
+  premiumIconSource,
+  onRuleItemClick = () => {},
+}) => {
+  const navigate = useNavigate();
+  const { getFeatureLimitValue } = useFeatureLimiter();
+
+  const handleRuleTypeClick = (ruleType: RuleType) => {
+    trackRuleCreationWorkflowStartedEvent(ruleType, source);
+
+    callback();
+
+    redirectToCreateNewRule(navigate, ruleType, source, groupId);
+  };
+
+  const checkIsPremiumRule = (ruleType: RuleType) => {
+    const featureName = `${ruleType.toLowerCase()}_rule`;
+    return !getFeatureLimitValue(featureName as FeatureLimitType);
+  };
+
+  return (
+    <div className="rule-selection-list-container">
+      {RULE_DETAILS.categories.map(({ title, type, rules }) => {
+        return (
+          <div key={type}>
+            <div className="rule-category-title">
+              <span className="title">{title}</span> <Divider plain />
+            </div>
+            <div>
+              {rules.map((rule) => {
+                return (
+                  <PremiumFeature
+                    source={source}
+                    featureName={`${rule.title} rule`}
+                    popoverPlacement={premiumPopoverPlacement}
+                    onContinue={() => handleRuleTypeClick(rule.type)}
+                    features={[`${rule.type.toLowerCase()}_rule` as FeatureLimitType, FeatureLimitType.num_rules]}
+                    onClickCallback={(e) => {
+                      e?.preventDefault?.();
+                      onRuleItemClick(rule.type);
+                    }}
```

Comment:
Please verify once that `onRuleItemClick` does not gets invoked, if premium popover shows up, else popup will disappear as well as the drawer gets unmounted